src/chat_module.ts

```diff
@@ -506,24 +512,76 @@ export class ChatModule implements ChatInterface {
           throw new Error("Last messages should be a string from the `user`.");
         }
         this.getPipeline().appendConversationMessage(
-          message.name ? message.name : roles[0],
+          Role.User,
           message.content,
         );
+        if (message.name !== undefined) {
+          this.getPipeline().overrideUserRoleName(message.name);
+        }
```

Comment:
I just realized these override methods we introduced (including my `overrideSystemPrompt()`) do irreversible changes to `llm_chat.ts`'s `this.conversation`.

e.g. if the `request1` uses a customized name, `request2` cannot use the default name anymore, even if the processing of `request1` and `request2` are supposedly functional.

For `RoleName`, can we still stick to the old
```typescript
  this.getPipeline().appendConversationMessage(
    message.name ? message.name : roles[1],
    message.content,
  );
```
Perhaps we can let `appendConversationMessage()` take in either `Role` or a customized string. This may be the easiest way to keep it stateless. Now that we think of it, we may also need to let `prefillStep()` optionally take in a customized name, since we let the last message be a regular input (before this PR the name is just ignored).

For the system prompt, perhaps let us save the default `.system` when initializing, and reset it in `resetChat()`.

`overrideFunctionCalling()` does not seem to have this issue since we do the checking every time we process the request.

Reply:
Thanks @CharlieFRuan. That makes sense! I updated appendConversationMessage to optionally take in a customized string. If it is not provided, we will use the role string from conv_template instead. And I updated `prefillStep()` in a similar way.